src/FAudio_gstreamer.c

```diff
@@ -444,21 +555,64 @@ uint32_t FAudio_GSTREAMER_init(FAudioSourceVoice *pSourceVoice, uint32_t type)
 	);
 
 	decoder_sink = gst_element_get_static_pad(decoder, "sink");
-	g_assert(decoder_sink);
-	g_assert(gst_pad_link(result->srcpad, decoder_sink) == GST_PAD_LINK_OK);
+	if (!decoder_sink)
+	{
+		LOG_ERROR(
+			pSourceVoice->audio,
+			"%s",
+			"Unable to get decoder sink pad"
+		)
+		goto free_with_bin;
+	}
+
+	if (gst_pad_link(result->srcpad, decoder_sink) != GST_PAD_LINK_OK)
+	{
+		LOG_ERROR(
+			pSourceVoice->audio,
+			"%s",
+			"Unable to get link our src pad to decoder sink pad"
+		)
+		gst_object_unref(decoder_sink);
+		goto free_with_bin;
+	}
+
 	gst_object_unref(decoder_sink);
 
-	g_assert(gst_element_link_many(
+	if (!gst_element_link_many(
 		result->resampler,
 		converter,
 		result->dst,
-		NULL
-	));
+		NULL))
+	{
+		LOG_ERROR(
+			pSourceVoice->audio,
+			"%s",
+			"Unable to get link pipeline"
+		)
+		goto free_with_bin;
+	}
 
 	/* send stream-start */
 	event = gst_event_new_stream_start("faudio/gstreamer");
-	g_assert(event);
-	g_assert(gst_pad_push_event(result->srcpad, event));
+	if (!event)
+	{
+		LOG_ERROR(
+			pSourceVoice->audio,
+			"Unable to create stream_start event; is %u-bit gstreamer installed?",
+			sizeof(void *) * 8
+		)
+		goto free_with_bin;
+	}
+
+	if (!gst_pad_push_event(result->srcpad, event))
+	{
```

Comment:
If this fails, ignore that. Only later when pushing buffers check the return value